camera_control/src/camera_control/camera_control_app.py

```diff
@@ -0,0 +1,114 @@
+        self.video_path = video_path
+
+        self.app_signals = CameraControlSignals()
+
+        self.setWindowTitle("Animal finder")
+        self.setGeometry(100, 100, 800, 600)
+
+        self.central_widget = QWidget(self)
+        self.setCentralWidget(self.central_widget)
+
+        # This label will display the video frames
+        self.frame_display = QLabel(self)
+        self.detection_mask_display = QLabel(self)
+
+        self.seek_slider = QSlider(Qt.Horizontal, self)
+        self.seek_slider.setMinimum(0)
+        self.seek_slider.setMaximum(100)
+        self.seek_slider.setValue(0)
+        self.seek_slider.valueChanged.connect(self.seek_frame)
+
+        default_detection_parameters = DetectionParameters(10, 11, "MOG2", 0.001)
+        self.detection_control_widget = DetectionControlWidget(self.app_signals, self.video_path, default_detection_parameters)
+
+        self.app_signals.frame_updated.connect(
+            lambda q_img: self.frame_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.detection_completed.connect(
+            lambda q_img: self.frame_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.detection_mask_updated.connect(
+            lambda q_img: self.detection_mask_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.frame_number_updated.connect(self.detection_control_widget.set_frame_number)
+
+        self.running = True
+                
+        self.cap = cv2.VideoCapture(self.video_path)
+        if not self.cap.isOpened():
+            raise ValueError(f"Failed to open video: {self.video_path}")
+        self.total_frames = int(self.cap.get(cv2.CAP_PROP_FRAME_COUNT))
+
+        self.layout = QHBoxLayout(self.central_widget)
+        self.video_layout = QVBoxLayout()
+        self.video_tabs = QTabWidget(self)
+        self.video_tabs.addTab(self.frame_display, "Input Video")
+        self.video_tabs.addTab(self.detection_mask_display, "Detection Mask")
+        self.video_layout.addWidget(self.video_tabs)
+        self.video_layout.addWidget(self.seek_slider)
+        self.layout.addLayout(self.video_layout)
+        self.side_panel_layout = QVBoxLayout()
+        self.side_panel_layout.addWidget(self.detection_control_widget)
+        self.layout.addLayout(self.side_panel_layout)
+
+        self.seek_slider.setValue(0)
+        # Manually trigger the first frame update.
+        self.seek_slider.valueChanged.emit(0)
+
+    def seek_frame(self, value):
+        frame_number = int((value / 100) * self.total_frames)
+        self.cap.set(cv2.CAP_PROP_POS_FRAMES, frame_number)
+
+        ret, frame = self.cap.read()
+        if ret:
+            frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+            height, width, num_channels = frame.shape
+            bytes_per_line = num_channels * width
+            q_img = QImage(
+                frame.data, width, height, bytes_per_line, QImage.Format_RGB888
```

Comment:
This gives me an error.  I think it's just that we are using different versions of PySide, based on this post: https://forum.qt.io/topic/154279/no-qimage-format_rgb888-pyqt6/2
I'm using PySide 6.7.2 (the latest one), and now that class is found in `QImage.Format.Format_RGB8881`

Reply:
I'm also on pyside 6.7.2, how odd! I'll look into this a bit more.

Reply:
This is what I see if I go to QIMage source. Do you see something similar?
<img width="1110" alt="image" src="https://github.com/user-attachments/assets/8ee17266-08b4-44bc-994e-dc8177d16954">

Reply:
Yep, exactly the same! I agree that `QImage.Format.Format_RGB888` is the right name to use. I'm still curious why this works though. I looked at `QImage.__getattribute__` in the debugger, and it is `<slot wrapper '__getattribute__' of 'Shiboken.Object' objects>` so there's some magic happening...... and indeed `getattr(QImage, "Format_RGB888")` works just fine, even though `Format_RGB888` isn't in `QIMage.__dict___`

Reply:
I'm guessing that they did some hack to make enum values appear at their parent scope to try and replicate the way things work in C